.github/workflows/assign-dependabot-reviewer.yml

```diff
@@ -0,0 +1,43 @@
+name: Assign Reviewer for Dependabot
+
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+    branches:
+      - main
+  issue_comment:
+    types: [ created ]
+
+jobs:
+  assign-reviewer:
+    # Only run if a dependabot PR on open or synchronize or the command /reassign-dependabot is commented in the PR
+    if: >
+      github.event.pull_request.user.login == 'dependabot[bot]' ||
+      (
+        github.event_name == 'issue_comment' && 
+        contains(github.event.comment.body, '/dependabot-reassign')
+      )
```

Comment:
Sucks that this is the only way to filter on author. This means the job will technically run on every PR, but at least it should fast-exit if the user doesn't match. It's noisy though.

Also noisy is my github action for [pinging infra on DMS Task edits](https://github.com/code-dot-org/code-dot-org/blob/a008c6e01cf236cc017c7a6ad60cb9165ad69391/.github/workflows/pr_check_for_manual_deploy_requirement.yml). I wonder if it would make sense to have a single complex action that handles this category of work? Probably premature, we can see how much we hate the noise.

Reply:
Since this is using a YAML if conditional, the workflow should be skipped completely if the PR wasn't opened by dependabot. I tested this by opening a PR and no action fires. However, if you issue the PR comment `/dependabot-reassign` it will trigger an action execution but it will be skipped after 3 seconds due to the conditional